### PR TITLE
Add SIGRTMAX and SIGRTMIN on linux-like systems

### LINKED ISSUE #1883: libc expose SIGRTMIN and SIGRTMAX
Exposing SIGRTMIN and SIGRTMAX might be problematic as they dynamically change and will therefore be hard to encode as constants. Perhaps exposing them in the form of functions is the best solution? I have very little experience porting from C to Rust so I am not sure what the best way of doing that would be. If someone gave me a point in the right direction then I would be willing to try to tackle this.

Target triplet: x86_64-linux-gnu

Man page for signals:
https://www.man7.org/linux/man-pages/man7/signal.7.html

Other crates looking for feature:
https://github.com/nix-rust/nix/issues/1143

### PR DESCRIPTION
**Pull request by mkroening**
Fixes #1883. This is based on https://github.com/rust-lang/libc/pull/2195, which has not seen activity in a while.

The original issue was resolved by making `__libc_current_sigrtmax` and `__libc_current_sigrtmax` `pub`.

I also added a test similar to `errqueue`'s, where the actual value of the macro is compared to `libc`'s implementation.

Closes #2195

### CODE REVIEW

File: src/unix/linux_like/mod.rs [hunk 1]
```diff
@@ -1362,6 +1362,16 @@ cfg_if! {
     }
 }
 
+safe_f! {
```
Comment (JohnTitor) [inline src/unix/linux_like/mod.rs:1365]: We have another `safe_f!` here, the below should be placed there. And we may want to make them `const fn`s.

File: src/unix/linux_like/mod.rs [hunk 2]
```diff
@@ -1480,6 +1490,9 @@ safe_f! {
 }
 
 extern "C" {
+    pub fn __libc_current_sigrtmax() -> ::c_int;
+    pub fn __libc_current_sigrtmin() -> ::c_int;
```
Comment (JohnTitor) [inline src/unix/linux_like/mod.rs:1494]: I think it's fine to expose but I'd like to make it `#[doc(hidden)]`.

**Review by JohnTitor — CHANGES_REQUESTED** [review]

File: src/unix/linux_like/mod.rs (hunk 2 above)
Comment (mkroening) [reply src/unix/linux_like/mod.rs]: Done. :+1:

File: src/unix/linux_like/mod.rs (hunk 1 above)
Comment (mkroening) [reply src/unix/linux_like/mod.rs]: I moved it to the other `safe_f!`, did I get the `const`ness right?

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks!
@bors r+

**Comment by mkroening** [comment]
[Looks](https://github.com/rust-lang/libc/runs/3196028707) like the `const`ness is not allowed here, since we call `__libc_current_sigrtmax`.

Should I remove the `const`ness again?

**Comment by JohnTitor** [comment]
Ah yes! I forgot that `__libc_*`s weren't `const`s.

**Comment by mkroening** [comment]
Done. :+1:

**Pull request merged**